src/SupabaseClient.ts

```diff
@@ -97,9 +97,9 @@ export default class SupabaseClient<
     this.headers = { ...DEFAULT_HEADERS, ...options?.headers }
     this.shouldThrowOnError = settings.shouldThrowOnError || false
 
+    this.auth = this._initSupabaseAuthClient(settings.auth || {}, this.headers, settings.fetch)
     this.fetch = fetchWithAuth(supabaseKey, this._getAccessToken.bind(this), settings.fetch)
 
-    this.auth = this._initSupabaseAuthClient(settings.auth || {}, this.headers, this.fetch)
```

Comment:
@soedirgo fyi this was a circular reference. Is there any reason why we can't use the regular `settings.fetch` here instead of our `fetchWithAuth`?

Reply:
looks like the circular reference is `this.auth` -> `this.fetch` -> `this._getAccessToken` -> `this.auth`

not quite familiar with gotrue-js but I can't think of any reason not to use `settings.fetch` off the top of my head (but we def can't use it for `functions` and `storage`)

Reply:
Correct. It's only if we were doing something extra in `fetchWithAuth` that `gotrue-js` needed... but yeah it just does a normal fetch so thanks, Bobbie!